### PR TITLE
Documentation mess :)

### PR DESCRIPTION
**Pull request by cyrillbolliger**
I've messed around regardless of anything. **Feel free to reject any suggested changes**!

You may want to have a look at the diffs of the individual commits...

### CODE REVIEW

File: handout.md [hunk 1]
```diff
@@ -8,6 +8,79 @@ classoption:
...
+## Configuration file: pom.xml
+The `pom.xml` is a **p**roject **o**bject **m**odel with xml format.
+
+Example `pom.xml`
+```xml
+<project>
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.mycompany.app</groupId>
+  <artifactId>my-app</artifactId>
+  <version>1</version>
```
Comment (martyschaer) [inline handout.md]: `<version>` should probably be according to https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN400

File: handout.md [hunk 2]
```diff
@@ -8,6 +8,79 @@ classoption:
...
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-core</artifactId>
+      <version>5.2.9.RELEASE</version>
+      </dependency>
+  </dependencies>
+</project>
+```
+
+While the dependencies are probably the most useful, but still optional feature of maven, the following properties are mandatory:
+ - `groupId`  - package path of the project
```
Comment (martyschaer) [inline handout.md:52]: An empty newline is required before lists for them to work (except directly following headings)

File: handout.md [hunk 3]
```diff
@@ -22,7 +95,7 @@ Git is an open source version control system (VCS),
 which was developed by Linus Torvalds in 2005
 because the current VCS used for Linux Kernel development
 was no longer sufficient.
-^[[git-scm.com A short history](https://git-scm.com/book/en/v2/Getting-Started-A-Short-History-of-Git)]
```
Comment (martyschaer) [inline handout.md:25]: This was intended as a footnote/reference.
I would leave this out entirely otherwise.

File: handout.md [hunk 4]
```diff
@@ -32,10 +105,33 @@ Some of the goals for Git were:
 * Efficiently handle large projects
 
 ## Quick Start Guide
+* Get and install git: [https://git-scm.com/downloads](https://git-scm.com/downloads)
```
Comment (martyschaer) [inline handout.md]: Prettify displayed links
```suggestion
* Get and install git: [https://git-scm.com/downloads](git-scm.com/downloads)
```

File: handout.md [hunk 5]
```diff
@@ -32,10 +105,33 @@ Some of the goals for Git were:
...
+* `git clone git@github.com/dude/project` - get a local copy of the project
+* `git fetch` - get the latest changes from the server, without applying them to your working copy
+* `git status` - list the changes on your working copy and show, if they are included in your next commit
+* `git add <pathspec>` - add the specified file (also multiple files and globing allowed) to the next commit (stage the file)
+* `git rm --cached <pathspec>` - remove the specified file from the commit list (unstage the file), but keep the local changes
```
Comment (martyschaer) [inline handout.md]:
```suggestion
* `git restore --staged <pathspec>` - remove the specified file from the commit list (unstage the file), but keep the local changes
```

`rm --cached` removes the specified file from the index.

File: handout.md [hunk 6]
```diff
@@ -32,10 +105,33 @@ Some of the goals for Git were:
...
+* `git checkout <branch>` - switch to the specified branch. Use `-b` to create a new branch
 
 ## Version Control
+Version control systems (VCS) serve mainly two purposes: 
+1) Keeping track of every change that happens to the code
+2) Facilitate simultaneous edition (by multiple programmers) of the same project
+
+Thanks to VCS it is possible to easily revert to any revision ever made, find out when exactly a feature (or bug) was introduced etc. Further more, it is also possible to see, who wrote what and when.
+
+The fantastic merging capabilities introduced with Git enable programmers to easily unify their working copies without overwriting each others changes. This outstanding feature made git the most popular VCS today. Still, every programmer has its own version, but git simplifies merging these versions into a new, combined one.
```
Comment (martyschaer) [inline handout.md]: It's usually more convenient to split long paragraphs or sentences over multiple lines (makes editing and diffing easier).
A new paragraph is only introduced with an entirely blank line.
A displayed newline within a paragraph can be achieved by adding two spaces (0x20) to the end of a line.

File: handout.md [hunk 7]
```diff
@@ -46,4 +142,7 @@ Some of the goals for Git were:
 \newpage
 ![](img/xkcd_1597.png)
 
-^[Image Credit: [xkcd.com/1597](https://xkcd.com/1597/)]
+[Image Credit: [xkcd.com/1597](https://xkcd.com/1597/)]
+
+![Branching model](img/git-branching-model.png)
```
Comment (martyschaer) [inline handout.md:181]: Image is not in the repo

File: handout.md [hunk 8]
```diff
@@ -32,10 +105,33 @@ Some of the goals for Git were:
...
+2) Facilitate simultaneous edition (by multiple programmers) of the same project
+
+Thanks to VCS it is possible to easily revert to any revision ever made, find out when exactly a feature (or bug) was introduced etc. Further more, it is also possible to see, who wrote what and when.
+
+The fantastic merging capabilities introduced with Git enable programmers to easily unify their working copies without overwriting each others changes. This outstanding feature made git the most popular VCS today. Still, every programmer has its own version, but git simplifies merging these versions into a new, combined one.
 
 ## Branching Model
+Apart of the very source of truth, the `main` (or `master`) branch, git allows to have some diverging source trees called branches. Branches allow to make and test some changes, without messing with the main code. A branch can be imagined as if it would be a completely separated copy of the code.
+
+It is very common to develop new features in a dedicated branch. This brings the advantage, that p.ex. an unfinished feature doesn't block the development of a hot-fix in the main branch.
```
Comment (martyschaer) [inline handout.md]:
```suggestion
It is very common to develop new features in a dedicated branch. This brings the advantage, that e.g. an unfinished feature doesn't block the development of a hot-fix in the main branch.
```

https://en.wiktionary.org/wiki/exempli_gratia

**Review by martyschaer — CHANGES_REQUESTED** [review]

File: handout.md (hunk 1 above)
Comment (cyrillbolliger) [reply handout.md]: absolutely. My bad, I just copied the snipped from https://maven.apache.org/guides/introduction/introduction-to-the-pom.html :see_no_evil:

File: handout.md (hunk 3 above)
Comment (cyrillbolliger) [reply handout.md:25]: ah sorry, I didn't knew the syntax for footnotes... I'll revert it.

Comment (cyrillbolliger) [reply handout.md:25]: hmmm, this somehow confuses syntax highlighting in vim...

File: handout.md (hunk 6 above)
Comment (cyrillbolliger) [reply handout.md]: I was just to lazy, but took `:%!fmt -80 -s` to the rescue.

File: handout.md (hunk 8 above)
Comment (cyrillbolliger) [reply handout.md]: thanks. learned something :100:

**Comment by cyrillbolliger** [comment]
Thanks for the review :1st_place_medal:

File: handout.md (hunk 3 above)
Comment (martyschaer) [reply handout.md:25]: That specific case works for my vim, but with large MD documents it does sometimes get confused.
Especially with cursive/bold font.

**Pull request merged**